File: src/main/java/com/graphbrain/db/Backend.java
package com.graphbrain.db;

import java.util.List;

public interface Backend {
	public abstract void close();
	public abstract Vertex get(String id, VertexType type);
	public abstract Vertex put(Vertex vertex);
	public abstract Vertex update(Vertex vertex);
	public abstract void remove(Vertex vertex);
	public abstract void associateEmailToUsername(String email, String username);
	public abstract String usernameByEmail(String email);
	public abstract List<Vertex> listByType(VertexType type);
}

File: core/src/main/scala/com/graphbrain/Backend.scala
package com.graphbrain.core

trait Backend {
	def close: Boolean
	def get(id: String, type: VertexType): Vertex
	def put(vertex: Vertex): Vertex
	def update(vertex: Vertex): Vertex
	def remove(vertex: Vertex)
	def associateEmailToUsername(email: String, username: String)
	def usernameByEmail(email: String): String
	def listByType(type: VertexType): List[Vertex]
	def edges(center: Vertex): Set[Edge]
}